### PR TITLE
Separate timeouts

### PR DESCRIPTION
**Pull request by michaelpanpliops**

### CODE REVIEW

File: server.cc [hunk 1]
```diff
@@ -160,7 +171,7 @@ RepStatus CheckpointProducer::WaitForCompletion(uint32_t timeout_msec)
     return producer_state_ == ProducerState::ERROR || producer_state_ == ProducerState::DONE;
```
Comment (tamarwo16) [inline server.cc:171]: IsFinalStage

**Review by tamarwo16 — CHANGES_REQUESTED** [review]

File: client.cc [hunk 2]
```diff
@@ -93,7 +98,10 @@ RepStatus GetStatus(RpcChannel& rpc, ServerState& state, uint64_t& num_kv_pairs,
 // Main entry function 
 // Kuaishou function: SyncManager::ReStoreFrom(const std::string &host, int32_t shard)->Status
 RepStatus ReplicateCheckpoint(RpcChannel& rpc, int32_t shard, const std::string &dst_path,
```
[the remark this answers is not in the document]
Comment (michaelpanpliops) [reply client.cc:100]: This is just a reference to the original Kuaishou code. I don't think it is relevant anymore. I will remove it.

File: client.cc [hunk 3]
```diff
@@ -42,7 +47,7 @@ RepStatus CheckpointConsumer::WaitForCompletion(uint32_t timeout_msec)
     return consumer_state_ == ConsumerState::ERROR || consumer_state_ == ConsumerState::DONE;
```
[the remark this answers is not in the document]
Comment (michaelpanpliops) [reply client.cc:47]: Good catch. I changed the stop function accordingly

File: pliops/consumer.cc [hunk 4]
```diff
@@ -45,7 +50,7 @@ void Consumer::WriterThread() {
     auto status = shard_->Put(wo, key, value);
     if(!status.ok()) {
       logger->Log(Severity::ERROR, FormatString("Writer thread: Failed inserting key %s, reason: %s\n", key, status.ToString()));
-      SetState(ConsumerState::ERROR, RepStatus(Code::NETWORK_FAILURE, Severity::ERROR, FormatString("Writer thread: Failed inserting key %s, reason: %s\n", key, status.ToString())));
+      SetState(ConsumerState::ERROR, RepStatus(Code::NETWORK_FAILURE, Severity::ERROR, FormatString("Writer thread: Failed inserting key %s, reason: %s", key, status.ToString())));
```
[the remark this answers is not in the document]
Comment (michaelpanpliops) [reply pliops/consumer.cc:53]: done

File: pliops/producer.cc [hunk 5]
```diff
@@ -154,7 +154,7 @@ void Producer::CommunicationThread() {
       shard_ = nullptr;
       if (!status.ok()) {
         logger->Log(Severity::ERROR, FormatString("Communication thread: shard_->Close failed, reason: %s\n", status.ToString()));
-        SetState(ProducerState::ERROR, RepStatus(Code::NETWORK_FAILURE, Severity::ERROR, FormatString("Communication thread: shard_->Close failed, reason: %s\n", status.ToString())));
+        SetState(ProducerState::ERROR, RepStatus(Code::NETWORK_FAILURE, Severity::ERROR, FormatString("Communication thread: shard_->Close failed, reason: %s", status.ToString())));
```
[the remark this answers is not in the document]
Comment (michaelpanpliops) [reply pliops/producer.cc:157]: done

File: server.cc (hunk 1 above)
Comment (michaelpanpliops) [reply server.cc:171]: done

File: client_main.cc [hunk 6]
```diff
@@ -69,13 +82,16 @@ int main(int argc, char* argv[]) {
   int threads;
```
[the remark this answers is not in the document]
Comment (michaelpanpliops) [reply client_main.cc:82]: I don't understand the question.

File: pliops/consumer.cc [hunk 7]
```diff
@@ -80,24 +85,24 @@ void Consumer::CommunicationThread()
 {
   logger->Log(Severity::INFO, FormatString("Communication thread started.\n"));
   std::unique_ptr<Connection<ConnectionType::TCP_SOCKET>> connection;
-  auto rc = Accept(*connection_, connection, timeout_msec_);
-  if (!rc.IsOk()) {
+  auto rc = Accept(*connection_, connection, connect_timeout_msec_);
```
Comment (william-cheung) [inline pliops/consumer.cc:88]: I think we need to differentiate accept timeout and socket r/w timeout.

Comment (michaelpanpliops) [reply pliops/consumer.cc:88]: This is exactly what connect_timeout_msec_ does

**Pull request merged**